src/retry.ts

```diff
@@ -0,0 +1,41 @@
+import * as core from '@actions/core'
+
+const maxRetries: number = parseInt(core.getInput('retries', { required: false }));
+
+export class Retries {
+	private maxAttemps: number
+
+	constructor() {
+		this.maxAttemps = maxRetries
+	}
+
+	async execute<T>(action: () => Promise<T>): Promise<T> {
+		// run N-1 attempts
+		var attempt = 1;
+		while (attempt < this.maxAttemps) {
+			try {
+				return await action()
```

Comment:
Was just gonna ask about the empty try/catch :slightly_smiling_face:

Reply:
Got too ahead of myself!